bip-0388.mediawiki

```diff
@@ -14,7 +14,10 @@
 
 == Abstract ==
 
-Wallet policies build on top of output script descriptors to represent the types of descriptors that are typically used to represent "accounts" in a software wallet, or a hardware signing device, in a compact, reviewable way. A wallet policy always represents exactly two descriptors, which produce the receive and change addresses that are logically part of the same account.
+Software wallets and hardware signing devices typically partition funds into separate "accounts". When signing or visualizing transactions, this allows to show to the user aggregate in-flow or out-flow information for one or more involved accounts.
```

Comment:
```suggestion
Software wallets and hardware signing devices typically feature "accounts" as a means to partition funds and transactions by purpose. This partitioning is also useful to show aggregate in-flow and out-flow information for the involved accounts when signing or visualizing transactions.
```

Reply:
I don't quite agree with the rewrite:
- "accounts"/wallet policies logically partition funds based on the spending conditions; therefore, it's mostly the ownership rules that is semantically represented in the "account". While _purpose_ has of course a lot of overlap with the ownership rules, I don't think it's the appropriate word to mention in this first paragraph.
- wallet policies / accounts partition funds, but don't quite _partition_ transactions. That might seem the case today for most wallets, but just because they only allows creating transactions spending from (and receiving change to) a single account. More sophisticated wallets could (and hopefully will) allow creating transactions that involve multiple accounts, including with different spending conditions or with different owners.